doc/design/volumes/nfsvolumestores.md

```diff
@@ -18,10 +21,36 @@ Allow the container user to
  2. Exposing shared storage configuration via VIC (e.g. IOPS per client, storage policy, etc.)
  3. Management of shared storage via VIC (e.g. container quiesce for storage maintenance, quota manipulation of the target, etc.)
 
+### Networking
+
+Containers will need to be on an appropriate network for the NFS volume store to be accessible.
+
+There are ways this could be done:
+ - allow association of volume-store with container-network to allow direct coupling
+ - note in the volume store list what network is required
+
+In our current networking model, this can potentially result in the container using the endpoint vm to NAT NFS traffic to/from the NFS target.  This is a potential bottleneck and single point of failure.  The other mode we support is adding an NFS container network, and then adding containers requiring the target to the same network.  This removes the point of failure but has other issues (*).
+
+_(*) Note: Without microsegmentation support, the services run in the container can potentially be exposed to the network the NFS target is on.  This means containers connecting to a specific NFS target all have direct connectivity to eachothers ports._
+
+Ultimately, (once we have microsegmentation support), we'd like to add the container to the appropriate container network in order for the container to have connectivity to the NFS target.
+
 ### Implementation
 
 Adding shared storage to our model fits with the `VolumeStore` interface.  At install, a VI admin can specify an NFS target as a `VolumeStore` (potentially) using a `nfs://host/<path>` URI with a volume store name.  And provided the VCH has access to the network the target is on, the container user only needs to pass the volume store name as one of the `volume create` driver opts to create a volume which will be backed by this shared storage target.  Then many containers can be created with the specified volume attached.
 
+#### Runtime network connectivity validation
+We need to inform the user when a container is being created without the appropriate network required to get connectivity to the NFS target.  We're planning to make changes to the personality such that when the container is created, it check if the specified container network matches the container network specified when the `VolumeStore` is created.
+
+This is just an example, but we could do the following:
+```
+$ vic-machine-linux create --volume-store volumeStoreName/nfs://host/path?<nfs container network name>
```

Comment:
Do we really need to explicitly associate the volume store with a container network? The vic user should make sure that the container is on the right network for the NFS store to successfully connect. Can we find that out during container init (in tether) and communicate this back during docker start/run?